### DISCUSSION TITLE
Target labels in Dynunet

### DISCUSSION THREAD

**Discussion by Joeycho** [Project-MONAI/tutorials#869] [Q&A] [answered]
Hello all,

I am also working toward for extracting network from nnUNet. I need more flexibility for testing modified versions.

What I found speicial in nnUNet is "Deep supervision" in training. I tracked it and I end up here with Dynunet.

I wonder how Dynunet deal with low-resolution target labels. nnUNet prepares target labels with quite long preprocessing. 

As nnUNet did it in loss calculation, I believe Dynunet should have low-resolution target labels, but I cannot find it clearly. Could you guide me to the parts of code?

https://github.com/MIC-DKFZ/nnUNet/blob/6d02b5a4e2a7eae14361cde9599bbf4ccde2cd37/nnunet/training/dataloading/dataset_loading.py#L379

**Comment by Joeycho** [comment, marked as answer]
https://github.com/Project-MONAI/tutorials/blob/8abcbc62c704af0bb22bedba617ae914c7802b97/modules/dynunet_pipeline/trainer.py#L56

I think, in the code, it keep the same target label and compare it with each prediction (different resolutions), but I am not sure that this corresponds with how nnUNet deal with it.

**Reply by yiheng-wang-nv** [reply]
to match up with the torchscript requirement, feature maps with different resolutions are resized into the target labels for calculation (see: https://github.com/Project-MONAI/MONAI/blob/fb976d7c6a2fc01640b3593775395f0a7f27b1db/monai/networks/nets/dynunet.py#L273), thus labels do not need to be resized into different resolutions. It does have a bit difference with nnUNet.

**Reply by Joeycho** [reply]
Ah, okay. That makes sense, too! Thanks for clarification!
